File: ch8/js/b-streaming-output.js
import { HumanMessage } from "@langchain/core/messages";

// Assuming graph is already created and configured
const graph = new StateGraph().compile();

const input = {
  messages: [
    new HumanMessage(
      "How old was the 30th president of the United States when he died?",
    ),
  ],
};

const config = { configurable: { thread_id: "1" } };

// Assuming graph is already created and configured
const output = await graph.stream(input, config);

for await (const chunk of output) {
  console.log(chunk);
}

File: ch8/py/b-streaming-output.py
from langchain_core.messages import HumanMessage
from langgraph.graph import StateGraph


def create_simple_graph():
    # Create a simple graph for demonstration
    builder = StateGraph()
    # Add nodes and edges as needed
    return builder.compile()


graph = create_simple_graph()

input = {
    "messages": [
        HumanMessage(
            "How old was the 30th president of the United States when he died?"
        )
    ]
}

for c in graph.stream(input, stream_mode="updates"):
    print(c)
